packages/transport-bluetooth/src/client/trezor-bluetooth.ts

```diff
@@ -0,0 +1,75 @@
+import { WebsocketClient } from '@trezor/websocket-client';
+
+import {
+    BluetoothDevice,
+    BluetoothInfo,
+    Logger,
+    NotificationEvent,
+    TrezorBluetoothSettings,
+} from './types';
+
+// Client for trezor-bluetooth rust websocket server
+export class TrezorBluetooth extends WebsocketClient<NotificationEvent> {
+    readonly settings: TrezorBluetoothSettings;
+    readonly logger: Logger;
+
+    constructor(settings: TrezorBluetoothSettings) {
+        super({
+            url: settings.url,
+            timeout: settings.timeout || 60 * 1000,
+        });
+        this.settings = Object.freeze(settings);
+        this.logger = settings.logger || {
+            info: () => {},
+            debug: () => {},
+            log: () => {},
+            warn: () => {},
+            error: () => {},
+        };
+    }
+
+    createWebsocket() {
+        return this.initWebsocket({
+            url: this.settings.url,
+            headers: {
+                Origin: 'https://node.trezor.io',
+                'User-Agent': 'Trezor Suite',
+            },
+            timeout: this.settings.timeout || 60 * 1000,
```

Comment:
line 19 and 38 look suspiciously same. are they both needed?

Reply:
well yes, both are needed `WebsocketClient` class naming is confusing.
`timeout` in WebsocketClient constructor (line 19) is used in `createDeferredManager` while `timeout` in `initWebsocket` (here) is used in `new WebSocket`

Reply:
removed [22ad13b](https://github.com/trezor/trezor-suite/pull/18214/commits/22ad13bb3846ad7fd0db2f7cce5db33c45dd8d2a) all unnecessary options, we can rely on defaults